src/middleware/builtin.ts

```diff
@@ -388,3 +401,21 @@ function isEventArgs(
 ): args is SlackEventMiddlewareArgs {
   return (args as SlackEventMiddlewareArgs).event !== undefined;
 }
+
+function tryTextMatching(
```

Comment:
I'm not sure about the best name for this private method yet. In particular, should we have `try` in the name? I think simpler names such as `matchText`, `matchWithStringOrRegeExp` are fine for this method. 

@stevengill @aoberoi @shaydewael 
Can you give a suggestion on the naming if there is something else in your mind? If `tryTextMatching` looks good to you, I think we can merge this PR as-is.

Reply:
The reason I avoided the `match-` prefix for this private method is because I worried `match-` prefixes could imply `public` (`matchMessage`, `matchEventType`, etc).

But I'm not sure about this too, so please feel free to suggest any names ☕